src/v/cloud_storage/remote_segment_index.cc

```diff
@@ -258,6 +265,12 @@ struct offset_index_header
     iobuf rp_index;
     iobuf kaf_index;
     iobuf file_index;
+
+    // Version 2 fields
+    int64_t base_time;
+    int64_t last_time;
+    std::vector<int64_t> time_write_buf;
+    iobuf time_index;
```

Comment:
do we need to care for version<1> ->version<2> compat?

Reply:
We're using serde's default read/write impls, & I believe `serde::envelope`'s default behavior is that if it just runs out of data on decode, it will not populate those later fields (so these values remain at their defaults).

Reply:
(I have also updated the base_time and last_time to have explicit initialization -- previously one would have had to use the vector or iobuf to check for whether timestamp fields are set, because the ints would have had undefined contents)